packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -221,7 +221,7 @@ export default class OptionManager {
 
     // resolve presets
     if (opts.presets) {
-      this.mergePresets(opts.presets, dirname);
+      this.mergePresets(opts.presets, dirname, opts.filename);
```

Comment:
@loganfsmyth possibly — it would need to be the closest `package.json` to `opts.filename`. The way it's done now will default to the previous behaviour and less likely to break how things work right now. Unsure how important that is in this case? Outside of that I'm happy for others to shuffle the code around to their preference :)